ui/src/app/workflows/components/workflow-details/workflow-details.tsx

```diff
@@ -352,6 +354,10 @@ export const WorkflowDetails = ({history, location, match}: RouteComponentProps<
         );
     };
     useEffect(() => {
+        // If a workflow is archived, we don't need watch.
+        if (isWfInDB) {
```

Comment:
We should do this here because `workflows.watch` calls `getWorkflow` in the serverside, which cause error described in #11373.

https://github.com/toyamagu-2021/argo-workflows/blob/b9e5e986c533571cacfc37ee2352c5e6860aa8dd/server/workflow/workflow_server.go#L233